# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/EarlyHintsService.h
# Commit: 8575cd92e1d8
# Full Hash: 8575cd92e1d8cd783e6cca3f6feadef11f0c4cc8
# Author: Manuel Bucher <manuel@mozilla.com>
# Date: 2022-12-03 09:24:59
# Regressor Bug: 1771867
# File Overlap Count: 1
# Description:
#   Bug 1771867 - Early Hints Phase 2 - Part 8: Add list of open channels from OngoingEarlyHints to RedirectToRealChannelArgs r=necko-reviewers,valentin,kershaw
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D161179
# ==============================================================================

diff -r ce8f0a8fa688 -r 8575cd92e1d8 netwerk/protocol/http/EarlyHintsService.h
--- a/netwerk/protocol/http/EarlyHintsService.h	Fri Dec 02 16:15:55 2022 +0000
+++ b/netwerk/protocol/http/EarlyHintsService.h	Fri Dec 02 16:15:56 2022 +0000
@@ -9,15 +9,17 @@
 #define mozilla_net_EarlyHintsService_h
 
 #include "mozilla/Maybe.h"
+#include "mozilla/RefPtr.h"
 #include "mozilla/TimeStamp.h"
 #include "nsStringFwd.h"
-#include "mozilla/RefPtr.h"
+#include "nsTArray.h"
 
 class nsIChannel;
 class nsIURI;
 
 namespace mozilla::net {
 
+class EarlyHintConnectArgs;
 class OngoingEarlyHints;
 
 class EarlyHintsService {
@@ -29,6 +31,9 @@
   void FinalResponse(uint32_t aResponseStatus);
   void Cancel();
 
+  void RegisterLinksAndGetConnectArgs(
+      nsTArray<EarlyHintConnectArgs>& aOutLinks);
+
  private:
   void CollectTelemetry(Maybe<uint32_t> aResponseStatus);
   void CollectLinkTypeTelemetry(const nsAString& aRel);
